src/components/flowchart/flowchart.test.js

```diff
@@ -281,6 +281,60 @@ describe('FlowChart', () => {
     ).toBe(2);
   });
 
+  it('does not apply pipeline-node--dataset-input class to input dataset nodes when not under focus mode', () => {
+    const wrapper = setup.mount(<FlowChart />);
+    expect(wrapper.render().find('.pipeline-node--dataset-input').length).toBe(
+      0
+    );
+  });
+
+  it('applies pipeline-node--dataset-input class to input dataset nodes under focus mode', () => {
+    const wrapper = setup.mount(
+      <FlowChart
+        nodeTypeDisabled={{ parameters: true }}
+        focusMode={{ id: 'pipeline1' }}
+        inputOutputDataNodes={{
+          '0ae9e4de': { id: '0ae9e4de', name: 'Elephant' },
+          '09f5edeb': { id: '09f5edeb', name: 'Bear' },
+        }}
+      />
+    );
+    expect(wrapper.render().find('.pipeline-node--dataset-input').length).toBe(
+      2
+    );
+  });
+
+  it('applies pipeline-edge--data--input class to input dataset edges under focus mode', () => {
```

Comment:
Nit: shouldn't this modifier be named as `pipeline-edge--dataset--input` to be consistent with `pipeline-node--dataset-input` (or the other way around)? Same comment about `pipeline-edge--parameters--input` and `pipeline-node--parameter-input` (although I have a feeling this naming is intentional because pipeline-edge might have more than 1 parameters, right?)

Reply:
sure, I can rename this as `pipeline-edge--dataset--input` to align this with the naming for the input nodes css class